src/app/auth/components/SignUpButtonGroup.tsx

```diff
@@ -1,17 +1,27 @@
+'use client'
 import { useRouter } from 'next/navigation'
 import BasicButton from '@/components/global/button/BasicButton'
 import { ButtonSize, ButtonTheme, ButtonVariant } from '@/components/global/button/button.types'
+import { useEffect, useState } from 'react'
 
 interface ISignUpButtonGroupProps {
   isNicknameChecked: boolean
-  handleSignUpComplete: () => void
+  handleSignUpComplete: (kakaoId: string) => void
 }
 
 export default function SignUpButtonGroup({ isNicknameChecked, handleSignUpComplete }: ISignUpButtonGroupProps) {
   const router = useRouter()
+  const [kakaoId, setKakaoId] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const params = new URLSearchParams(window.location.search)
+      setKakaoId(params.get('kakaoId'))
```

Comment:
next.js에서 지원하는 훅이 있어요 !!

```suggestion
      const kakaoId = useSearchParams().get('kakaoId')
      setKakaoId(params.get(kakaoId))
```

Reply:
넴! 맨 처음에는 `useSearchParams` 활용해서 작성했는데 빌드 에러 나서 수정했습니다 ㅠㅠㅠ !!